e2e/pages/Table/TableModule.ts

```diff
@@ -231,22 +259,50 @@ class TableModule {
     await this.bulkImportRegistrationsButton.getByRole('button').click();
   }
 
-  async validateBulkActionTargetedPasNumber(expectedNumber: number) {
+  async validateBulkActionTargetedPasNumber({
+    expectedNumber,
+    bulkAction,
+  }: {
+    expectedNumber: number;
+    bulkAction: string;
+  }) {
     const textLocator = this.page
       .locator('p')
-      .filter({ hasText: 'Send Message to PAs' });
-    const textContent = await textLocator.textContent();
-
-    if (textContent !== null) {
-      const expectedText = `Send Message to PAs for ${expectedNumber} People Affected.`;
-      if (textContent.trim() !== expectedText) {
-        throw new Error(
-          `Expected "${expectedText}" but received "${textContent.trim()}"`,
-        );
+      .filter({ hasText: sendMessageAction });
+    const expectedText = `${sendMessageAction} for ${expectedNumber} People Affected.`;
+
+    if (
+      !(await this.retryCheckTextContentOfBulkAction({
+        textLocator: textLocator,
+        expectedText: expectedText,
+        bulkAction: bulkAction,
+      }))
+    ) {
+      expect(expectedText).toBe(await textLocator.textContent());
+    }
+  }
+
+  async retryCheckTextContentOfBulkAction({
```

Comment:
I think this function could be a bit clearer using an interface as input

Reply:
Done